server/CVCMIServer.cpp

```diff
@@ -615,9 +625,24 @@ int main(int argc, char** argv)
 		//and return non-zero status so client can detect error
 		throw;
 	}
+#ifdef VCMI_ANDROID
+	AndroidVMHelper envHelper;
+	envHelper.callStaticVoidMethod(AndroidVMHelper::NATIVE_METHODS_DEFAULT_CLASS, "killServer");
+#endif
 	delete VLC;
 	VLC = nullptr;
 	CResourceHandler::clear();
 
   return 0;
 }
+
+#ifdef VCMI_ANDROID
+
+void CVCMIServer::create()
+{
+	boost::this_thread::sleep(boost::posix_time::milliseconds(1000));

```

Comment:
This I can answer. There is no boost::interprocess on Android so this is used instead. In fact we could actually benefit from removing interprocess everywhere as well since it's the only thing we using it for. I think we discussed that.